NodeUpdater: provide info on file name when JSON parsing fails

When the `package.json` parsing fails for some reason, NodeUpdater should wrap the exception and provide the file name of the json file which failed to get parsed.

The stacktrace applies to Vaadin 14, but the same issue is present in Vaadin 20. It is only necessary to fix this on Vaadin 20.

Stacktrace:
```
Caused by: elemental.json.JsonException: Invalid object: 
	at elemental.json.impl.JsonTokenizer.parseObject(JsonTokenizer.java:275)
	at elemental.json.impl.JsonTokenizer.nextValue(JsonTokenizer.java:173)
	at elemental.json.impl.JsonTokenizer.parseObject(JsonTokenizer.java:231)
	at elemental.json.impl.JsonTokenizer.nextValue(JsonTokenizer.java:173)
	at elemental.json.impl.JsonTokenizer.parseObject(JsonTokenizer.java:231)
	at elemental.json.impl.JsonTokenizer.nextValue(JsonTokenizer.java:173)
	at elemental.json.impl.JreJsonFactory.parse(JreJsonFactory.java:64)
	at elemental.json.Json.parse(Json.java:53)
	at com.vaadin.flow.server.frontend.NodeUpdater.getJsonFileContent(NodeUpdater.java:209)
	at com.vaadin.flow.server.frontend.NodeUpdater.getPackageJson(NodeUpdater.java:192)
	at com.vaadin.flow.server.frontend.TaskCreatePackageJson.execute(TaskCreatePackageJson.java:47)
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:512)
	at com.vaadin.gradle.VaadinPrepareFrontendTask.vaadinPrepareFrontend(VaadinPrepareFrontendTask.kt:107)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
```
### Versions:
    - Vaadin / Flow version: Vaadin 14.x, 20.x

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
@@ -37,6 +37,7 @@ import com.vaadin.flow.server.frontend.scanner.FrontendDependencies;
 import com.vaadin.flow.server.frontend.scanner.FrontendDependenciesScanner;
 
 import elemental.json.Json;
+import elemental.json.JsonException;
 import elemental.json.JsonObject;
 import elemental.json.JsonValue;
 
@@ -244,7 +245,14 @@ public abstract class NodeUpdater implements FallibleCommand {
         if (packageFile.exists()) {
             String fileContent = FileUtils.readFileToString(packageFile,
                     UTF_8.name());
-            jsonContent = Json.parse(fileContent);
+            try {
+                jsonContent = Json.parse(fileContent);
+            } catch (JsonException e) {
+                throw new RuntimeException(
+                        "Cannot parse package file located in path: "
+                                + packageFile,
+                        e);
+            }
         }
         return jsonContent;
     }

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
@@ -248,10 +248,8 @@ public abstract class NodeUpdater implements FallibleCommand {
             try {
                 jsonContent = Json.parse(fileContent);
             } catch (JsonException e) {
-                throw new RuntimeException(
-                        "Cannot parse package file located in path: "
-                                + packageFile,
-                        e);
+                throw new JsonException(String.format(
+                        "Cannot parse package file '%s'", packageFile));
             }
         }
         return jsonContent;

--- a/flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdaterTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdaterTest.java
@@ -25,7 +25,6 @@ import java.util.stream.Stream;
 import org.apache.commons.io.FileUtils;
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.core.StringContains;
-import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Rule;
@@ -38,6 +37,7 @@ import com.vaadin.flow.server.frontend.scanner.ClassFinder;
 import com.vaadin.flow.server.frontend.scanner.FrontendDependencies;
 
 import elemental.json.Json;
+import elemental.json.JsonException;
 import elemental.json.JsonObject;
 import static com.vaadin.flow.server.Constants.COMPATIBILITY_RESOURCES_FRONTEND_DEFAULT;
 import static com.vaadin.flow.server.Constants.RESOURCES_FRONTEND_DEFAULT;
@@ -59,8 +59,6 @@ public class NodeUpdaterTest {
 
     private URL url;
 
-    private File brokenPackageJsonFile;
-
     @Before
     public void setUp() throws IOException {
         url = new URL("file://bar");
@@ -76,16 +74,6 @@ public class NodeUpdaterTest {
             }
 
         };
-        brokenPackageJsonFile = temporaryFolder.newFile("broken-package.json");
-        FileUtils.writeStringToFile(brokenPackageJsonFile,
-                "{ some broken json ", UTF_8);
-    }
-
-    @After
-    public void tearDown() throws IOException {
-        npmFolder.delete();
-        resourceFolder.delete();
-        brokenPackageJsonFile.delete();
     }
 
     @Test
@@ -231,13 +219,19 @@ public class NodeUpdaterTest {
     }
 
     @Test
-    public void getJsonFileContent_incorrectPackageJsonContent_throwsExceptionWithFileName() {
-        RuntimeException exception = Assert.assertThrows(RuntimeException.class,
+    public void getJsonFileContent_incorrectPackageJsonContent_throwsExceptionWithFileName()
+            throws IOException {
+        File brokenPackageJsonFile = temporaryFolder
+                .newFile("broken-package.json");
+        FileUtils.writeStringToFile(brokenPackageJsonFile,
+                "{ some broken json ", UTF_8);
+
+        JsonException exception = Assert.assertThrows(JsonException.class,
                 () -> NodeUpdater.getJsonFileContent(brokenPackageJsonFile));
 
         MatcherAssert.assertThat(exception.getMessage(),
                 StringContains.containsString(
-                        "Cannot parse package file located in path: "));
+                        "Cannot parse package file "));
         MatcherAssert.assertThat(exception.getMessage(),
                 StringContains.containsString("broken-package.json"));
     }

--- a/flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdaterTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdaterTest.java
@@ -22,6 +22,9 @@ import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import org.apache.commons.io.FileUtils;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.StringContains;
 import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
@@ -39,6 +42,7 @@ import elemental.json.JsonObject;
 import static com.vaadin.flow.server.Constants.COMPATIBILITY_RESOURCES_FRONTEND_DEFAULT;
 import static com.vaadin.flow.server.Constants.RESOURCES_FRONTEND_DEFAULT;
 import static com.vaadin.flow.server.frontend.FrontendUtils.FLOW_NPM_PACKAGE_NAME;
+import static java.nio.charset.StandardCharsets.UTF_8;
 
 public class NodeUpdaterTest {
 
@@ -55,6 +59,8 @@ public class NodeUpdaterTest {
 
     private URL url;
 
+    private File brokenPackageJsonFile;
+
     @Before
     public void setUp() throws IOException {
         url = new URL("file://bar");
@@ -70,12 +76,16 @@ public class NodeUpdaterTest {
             }
 
         };
+        brokenPackageJsonFile = temporaryFolder.newFile("broken-package.json");
+        FileUtils.writeStringToFile(brokenPackageJsonFile,
+                "{ some broken json ", UTF_8);
     }
 
     @After
     public void tearDown() throws IOException {
         npmFolder.delete();
         resourceFolder.delete();
+        brokenPackageJsonFile.delete();
     }
 
     @Test
@@ -218,6 +228,18 @@ public class NodeUpdaterTest {
         JsonObject packageJson = NodeUpdater.getJsonFileContent(
                 formPackageJsonFile);
         Assert.assertEquals("bar", packageJson.getString("foo"));
+    }
+
+    @Test
+    public void getJsonFileContent_incorrectPackageJsonContent_throwsExceptionWithFileName() {
+        RuntimeException exception = Assert.assertThrows(RuntimeException.class,
+                () -> NodeUpdater.getJsonFileContent(brokenPackageJsonFile));
+
+        MatcherAssert.assertThat(exception.getMessage(),
+                StringContains.containsString(
+                        "Cannot parse package file located in path: "));
+        MatcherAssert.assertThat(exception.getMessage(),
+                StringContains.containsString("broken-package.json"));
     }
 
     private String getPolymerVersion(JsonObject object) {

